txnsync/txngroups.go

```diff
@@ -17,23 +17,417 @@
+	BitmaskAssetParams bitmask              `codec:"aparbm"`
+}
+
+type encodedAssetTransferTxnFields struct {
+	_struct struct{} `codec:",omitempty,omitemptyarray"`
+
+	XferAsset        []basics.AssetIndex `codec:"xaid,allocbound=maxEncodedTransactionGroup"`
+	BitmaskXferAsset bitmask             `codec:"xaidbm"`
+
+	AssetAmount        []uint64 `codec:"aamt,allocbound=maxEncodedTransactionGroup"`
+	BitmaskAssetAmount bitmask  `codec:"aamtbm"`
+
+	AssetSender        []byte  `codec:"asnd,allocbound=maxAddressBytes"`
+	BitmaskAssetSender bitmask `codec:"asndbm"`
+
+	AssetReceiver        []byte  `codec:"arcv,allocbound=maxAddressBytes"`
+	BitmaskAssetReceiver bitmask `codec:"arcvbm"`
+
+	AssetCloseTo        []byte  `codec:"aclose,allocbound=maxAddressBytes"`
+	BitmaskAssetCloseTo bitmask `codec:"aclosebm"`
+}
+
+type encodedAssetFreezeTxnFields struct {
+	_struct struct{} `codec:",omitempty,omitemptyarray"`
+
+	FreezeAccount        []byte  `codec:"fadd,allocbound=maxAddressBytes"`
+	BitmaskFreezeAccount bitmask `codec:"faddbm"`
+
+	FreezeAsset        []basics.AssetIndex `codec:"faid,allocbound=maxEncodedTransactionGroup"`
+	BitmaskFreezeAsset bitmask             `codec:"faidbm"`
+
+	AssetFrozen        []bool  `codec:"afrz,allocbound=maxEncodedTransactionGroup"`
+	BitmaskAssetFrozen bitmask `codec:"afrzbm"`
+}
+
+//msgp:allocbound applicationArgs transactions.EncodedMaxApplicationArgs
+type applicationArgs [][]byte
+
+//msgp:allocbound addresses transactions.EncodedMaxAccounts
+type addresses []basics.Address
+
+//msgp:allocbound appIndeces transactions.EncodedMaxForeignApps
+type appIndeces []basics.AppIndex
+
+//msgp:allocbound assetIndeces transactions.EncodedMaxForeignAssets
+type assetIndeces []basics.AssetIndex
+
+//msgp:allocbound program config.MaxAppProgramLen
+type program []byte
+
+type encodedApplicationCallTxnFields struct {
+	_struct struct{} `codec:",omitempty,omitemptyarray"`
+
+	ApplicationID        []basics.AppIndex `codec:"apid,allocbound=maxEncodedTransactionGroup"`
+	BitmaskApplicationID bitmask           `codec:"apidbm"`
+
+	OnCompletion        []transactions.OnCompletion `codec:"apan,allocbound=maxEncodedTransactionGroup"`
+	BitmaskOnCompletion bitmask                     `codec:"apanbm"`
+
+	ApplicationArgs        []applicationArgs `codec:"apaa,allocbound=maxEncodedTransactionGroup"`
+	BitmaskApplicationArgs bitmask           `codec:"apaabm"`
+
+	Accounts        []addresses `codec:"apat,allocbound=maxEncodedTransactionGroup"`
+	BitmaskAccounts bitmask     `codec:"apatbm"`
+
+	ForeignApps        []appIndeces `codec:"apfa,allocbound=maxEncodedTransactionGroup"`
+	BitmaskForeignApps bitmask      `codec:"apfabm"`
+
+	ForeignAssets        []assetIndeces `codec:"apas,allocbound=maxEncodedTransactionGroup"`
+	BitmaskForeignAssets bitmask        `codec:"apasbm"`
+
+	LocalStateSchema        []basics.StateSchema `codec:"apls,allocbound=maxEncodedTransactionGroup"`
```

Comment:
you'll need to break that into it's component to avoid the keys appear over and over in the encoded form.